Content.Shared/SubFloor/TrayScanRevealSystem.cs

```diff
@@ -0,0 +1,44 @@
+using System.Linq;
+using Robust.Shared.Map.Components;
+
+namespace Content.Shared.SubFloor;
+
+public sealed class TrayScanRevealSystem : EntitySystem
+{
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly SharedSubFloorHideSystem _subFloorHide = default!;
+    [Dependency] private readonly SharedMapSystem _map = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TrayScanRevealComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<TrayScanRevealComponent, ComponentRemove>(OnRemove);
+    }
+
+    private void OnStartup(EntityUid uid, TrayScanRevealComponent component, EntityEventArgs _)
+    {
+        var gridUid = _transform.GetGrid(uid);
+        if (gridUid is null)
+            return;
+
+        var gridComp = Comp<MapGridComponent>(gridUid.Value);
+        var position = _transform.GetGridOrMapTilePosition(uid);
+
+        (component.TileGridUid, component.TileGridComp, component.TileIndices) = ((EntityUid)gridUid, gridComp, position);
+        _subFloorHide.UpdateTile((EntityUid)gridUid, gridComp, position);
+    }
+
```

Comment:
This is extremely expensive and server doesn't care about subfloor so this shouldn't be in shared, nor does this account for a lot of things properly like movement.

Reply:
I considered the movement issue. But all the considered entities (like catwalk, carpets) are meant to be never unanchorable by any means so they won't move, even when bombed.